src/AAVE/AaveFlashLoan.sol

```diff
@@ -0,0 +1,97 @@
+pragma solidity 0.8.20;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "forge-std/console.sol";
+import {
+  IFlashLoanSimpleReceiver,
+  IPoolAddressesProvider,
+  IPool
+} from "./interfaces/IFlashLoanSimpleReceiver.sol";
+import {CErc20Interface,CTokenInterface} from "../CTokenInterfaces.sol";
+import {ISwapRouter} from "./interfaces/ISwapRouter.sol";
+import {CErc20Delegator} from "../CErc20Delegator.sol";
+
+// TODO: Inherit IFlashLoanSimpleReceiver
+contract AaveFlashLoan {
+
+  address admin;
+  address constant uniswapV3Router = 0xE592427A0AEce92De3Edee1F18E0157C05861564;
+  struct CompoundData { 
+   address borrower;
+   uint cRepayAmount;
+   address cTokenCBorrow;
+   address cTokenCollateral;
+   address underlyTokenCollateral;
+   address aavePool;
+  }
+
+  constructor(address _admin){
+    admin = _admin;
+  }
+ 
+  
+ function execute(uint256 aaveAmount,uint cRepayAmount, address aaveAsset,address aavePool
+    , address cTokenCBorrow , address cTokenCollateral, address underlyTokenCollateral ,address borrower) external {
+    // TODO
+
+    //組參數
+    CompoundData memory compoundData;
+    compoundData.borrower=borrower;
+    compoundData.cTokenCBorrow =cTokenCBorrow;
+    compoundData.cTokenCollateral =cTokenCollateral;
+    compoundData.aavePool=aavePool;
+    compoundData.cRepayAmount=cRepayAmount;
+    compoundData.underlyTokenCollateral = underlyTokenCollateral;
+    bytes memory params = abi.encode(compoundData);
+    
+    //aave flashLoan
+    POOL(aavePool).flashLoanSimple(address(this), aaveAsset, aaveAmount, params, 0);
+  }
+
+  //aave call back
+  function executeOperation(address asset, uint256 amount, uint256 premium, address initiator, bytes calldata params) external returns (bool) {
```

Comment:
You should check 2 things here:
1. msg.sender needs to be the aave pool
2. initiator is admin address